src/checkers/inference/solver/backend/maxsat/MaxSatSolver.java

```diff
@@ -43,7 +40,7 @@
  * @author jianchu
  *
  */
-public class MaxSatSolver extends Solver<MaxSatFormatTranslator> {
+public class MaxSatSolver extends ExternalSolver<MaxSatFormatTranslator> {
```

Comment:
Check whether external solver needs to be invoked? Or should this class be abstract?

Reply:
MaxSatSolver uses an internal instantiation of Sat4J, whereas LingeLing instantiates the external solver. The two solvers share common encoding and decoding logic.

I think it may be better to change ExternalSolver into a static helper class; the internal and external versions of z3 share some common logic as well.